st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -92,6 +125,50 @@ def _get_default(_actionchain):
         return _actionchain.chain[0].name
 
     @staticmethod
+    def _get_all_nodes(action_chain):
```

Comment:
Models should have no notion of how they are consumed. Utilities of these sort belong out side of model classes since the meaning and interaction should necessarily be handled outside of the model class. This way arguably different services could interpret the same model in a different way - unlikely but possible.

e.g. 
https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/stormbase.py#L79
and
https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/stormbase.py#L90

Do not have to belong to this type. They are operations that are closely related but not a feature of the type itself.

ONTH https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/stormbase.py#L223

the mixin approach is somewhat different pattern since it is the only purpose of the mixin is to augment with specific functionality. It is arguable if mixins should be used in the first place with models as they are better suited with service classes.

Model classes hold data and service classes perform business logic.

You could have a pragmatist argument about this with and say how else is one supposed to interpret the ActionChain model and I would counter with well what else is supposed to interpret the model? Also, if there is a need to perform such a validation then it can easily be done in a standalone actionchain validator.